src/librustc_parse/parser/pat.rs

```diff
@@ -295,6 +295,21 @@ impl<'a> Parser<'a> {
             // A rest pattern `..`.
             self.bump(); // `..`
             PatKind::Rest
+        } else if self.check(&token::DotDotDot) && !self.is_pat_range_end_start(1) {
+            // A rest pattern `...`.
+            self.bump(); // `...`
+
+            // A ... is not expected here
+            self.struct_span_err(lo, "unexpected `...`")
+                .span_label(lo, "not expected here")
+                .span_suggestion(
+                    self.token.span,
```

Comment:
```suggestion
                    lo,
```
You have already advanced the cursor, so in e.g. `... |` we would have `self.token.span` pointing at `|` and not `...`.
(For the previous span you can also use `self.prev_token.span`, but `lo` is equivalent here.)

Reply:
Got it. Thanks